src/main/app/form/Form.ts

```diff
@@ -11,16 +11,20 @@ export class Form {
     const parsedBody = Object.entries(this.form.fields)
       .map(([key, field]) => {
         if ((field as FormOptions)?.type === 'checkboxes') {
-          field.parser = formData => {
-            const checkbox = formData[key] as string[];
-            return checkbox[checkbox.length - 1];
-          };
+          field.parser = formData =>
+            (field as FormOptions).values.reduce((previous, currentCheckbox) => {
+              const checkboxName = currentCheckbox.name as string;
+              const checkboxValue = formData[checkboxName] as string[];
+              return [...previous, [checkboxName, checkboxValue[checkboxValue.length - 1]]];
+            }, [] as string[][]);
         }
-
         return [key, field];
       })
       .filter(([, field]) => typeof (field as FormOptions)?.parser === 'function')
-      .map(([key, field]) => [key, (field as FormOptions)?.parser?.(body)]);
+      .flatMap(([key, field]) => {
+        const parsed = (field as FormOptions).parser?.(body);
+        return Array.isArray(parsed) ? parsed : [[key, parsed]];
+      });
```

Comment:
this whole section of code is too complicated now. break it down into parts with names (or make it simpler)

Reply:
👍 I've refactored this into a seperate method in 0f6aed0